lib/Serialization/Serialization.cpp

```diff
@@ -579,10 +579,11 @@ void Serializer::writeHeader(const SerializationOptions &options) {
     ModuleName.emit(ScratchRecord, M->getName().str());
 
     SmallString<32> versionStringBuf;
+    auto version = M->getASTContext().LangOpts.EffectiveLanguageVersion;
     llvm::raw_svector_ostream versionString(versionStringBuf);
-    versionString << Version::getCurrentLanguageVersion();
+    versionString << version;
```

Comment:
This is used in diagnostics to describe module format mismatches, so I think we have no choice but to print the new version. (Or use a more complicated string.)